src/modules/UI/components/ErrorAlert/actions.js

```diff
@@ -1,13 +1,25 @@
 // @flow
+import { Alert } from 'react-native'
+
+import * as actions from '../../../../actions/indexActions.js'
+import s from '../../../../locales/strings.js'
+import type { Dispatch } from '../../../../modules/ReduxTypes'
 
 const PREFIX = 'UI/components/ErrorAlert/'
 export const DISPLAY_ERROR_ALERT = PREFIX + 'DISPLAY_ERROR_ALERT'
 export const DISMISS_ERROR_ALERT = PREFIX + 'DISMISS_ERROR_ALERT'
 
-export const displayErrorAlert = (message: string) => ({
+export const displayErrorAlertStore = (message: string) => ({
   type: DISPLAY_ERROR_ALERT,
```

Comment:
What is the purpose of creating a new function here that only has one special case?

Reply:
Basically flow... So `displayErrorAlert` still exists and it is now a thunk. This enables me to catch the error that we need to change and give it better text. However, in `src/lib/errorAlert` lives a function that would display any error dispatching it through the store. It would use the old Action creator `displayErrorAlert`.  Once `displayErrorAlert` became a thunk, flow was not remotely happy about it in the `errorAlert.js` file.  

`displayErrorAlert` is called throughout the code base. This is why I was changed it to a thunk where we could filter the message. errorAlert.js lives outside of redux and needed to just dispatch a simple action. By creating a new action for that that did exactly what the old one did, I was able to add my filtering and make it work.